Default disclosure flag to true for non-standard FOIs

willmcb opened the pull request:
Defaults to flag for disclosure specialist to true, and then hides these
options from the user when the case is non standard.

## Self-review checklist
<!-- Action these things before requesting reviews -->
* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Related JIRA tickets
[CT-2317](https://dsdmoj.atlassian.net/browse/CT-2317)

### Manual testing instructions
1. create case > foi
2. for type select either of the non-standard options, you should see that the "Flag for disclosure specialist" radio option at the bottom of the page should hide itself. If you click standard case type it should show itself again.
3. create the non-standard case. It should have automatically been flagged for a disclosure specialist.

digitalali-moj reviewed (approved):
JS looks good to me - haven't been able to test but I'm sure you've done that. It but make sense logically. Nice work. Just a reminder to please test on a few browsers, which you probably doing anyway.

With regards to the hiding of elements - as long as it's not essential since some modes may still show it if it's CSS hidden. e.g. non style view for Accessibility. So there is a small chance of changing it in that view - but not sure if it's an  issue, maybe worth double checking.

Commit:
Defaults to flag for disclosure specialist to true, and then hides these
options from the user when the case is non standard.

Commit:
Add tests

Commit:
Fix specs

Commit:
Delete spec as it is no longer relevant

All complaince cases should be trigger cases now so this spec
is not needed anymore.
